pkg/cmd/cli/cmd/newapp.go

```diff
@@ -543,7 +542,7 @@ func setAnnotations(annotations map[string]string, result *newcmd.AppResult) err
 	for _, object := range result.List.Items {
 		err := util.AddObjectAnnotations(object, annotations)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to add annotation to object of type %q, this type is probably unknown.", object.GetObjectKind().GroupVersionKind())
```

Comment:
how about:
`failed to add annotation to object of type %q, this resource type is probably unsupported by your client version.`

(I hate the probably, but w/o making the code a lot messier there's no way to be *certain* that is what went wrong here)